extension/healthcheckextension/healthcheckextension.go

```diff
@@ -55,21 +64,79 @@ func (hc *healthCheckExtension) Start(_ context.Context, host component.Host) er
 		return err
 	}
 
-	// Mount HC handler
-	hc.server.Handler = hc.state.Handler()
-	hc.stopCh = make(chan struct{})
-	go func() {
-		defer close(hc.stopCh)
+	if !hc.config.MetricsHealthCheck.Enabled {
+		// Mount HC handler
+		hc.server.Handler = hc.state.Handler()
+		hc.stopCh = make(chan struct{})
+		go func() {
+			defer close(hc.stopCh)
+
+			// The listener ownership goes to the server.
+			if err := hc.server.Serve(ln); err != http.ErrServerClosed && err != nil {
+				host.ReportFatalError(err)
+			}
+		}()
+	} else {
+		err = hc.initExporter()
```

Comment:
This approach doesn't seem scalable to more health check conditions - we should instead just have one, our current handler, and it can go through optional healthcheckers that report a health status. It would return an OK response only if all of these healthcheckers reported healthy.

@jpkrohling You had some thoughts on this, can you provide some guidance on how to structure this addition to the healthcheck?

Reply:
actually the current handler is defined in another jaeger package which i cannot change it, that's why i can only create my own handler.

Reply:
My idea was to move away from the Jaeger health check in favor of a proper lib for this. I haven't used, but one that I would give a try is this one: https://github.com/hellofresh/health-go

Reply:
Another question is that can we make this an out of scope goal of this pr? we can have another further modification of the health check functions.